sentry/src/db/event_aggregate.rs

```diff
@@ -1,38 +1,72 @@
 use crate::db::DbPool;
 use bb8::RunError;
-use bb8_postgres::tokio_postgres::types::ToSql;
+use bb8_postgres::tokio_postgres::binary_copy::BinaryCopyInWriter;
+use bb8_postgres::tokio_postgres::types::{ToSql, Type};
+use bb8_postgres::tokio_postgres::Error;
 use chrono::{DateTime, Utc};
+use futures::pin_mut;
 use primitives::sentry::EventAggregate;
-use primitives::ValidatorId;
+use primitives::BigNum;
+use primitives::{ChannelId, ValidatorId};
+use std::ops::Add;
 
 pub async fn list_event_aggregates(
     pool: &DbPool,
+    channel_id: &ChannelId,
     limit: u32,
     from: &Option<ValidatorId>,
     after: &Option<DateTime<Utc>>,
 ) -> Result<Vec<EventAggregate>, RunError<bb8_postgres::tokio_postgres::Error>> {
     let (mut where_clauses, mut params) = (vec![], Vec::<&(dyn ToSql + Sync)>::new());
+    let id = channel_id.to_string();
+    where_clauses.push(format!("channel_id = '{}'", id));
+
     if let Some(from) = from {
-        let key_counts = format!(
-            "events->'IMPRESSION'->'eventPayouts'->'{}'",
-            from.to_string()
-        );
-        where_clauses.push(format!("{} IS NOT NULL", key_counts));
+        where_clauses.push(format!("earner = '{}'", from.to_string()));
+        where_clauses.push("event_type = 'IMPRESSION'".to_string());
+    } else {
+        where_clauses.push("earner is NOT NULL".to_string());
     }
+
     if let Some(after) = after {
         params.push(after);
-        where_clauses.push(format!("created > {}", params.len()));
+        where_clauses.push(format!("created > ${}", params.len()));
     }
 
     let event_aggregates = pool
         .run(move |connection| {
             async move {
                 let where_clause = if !where_clauses.is_empty() {
-                    format!("WHERE {}", where_clauses.join(" AND "))
+                    where_clauses.join(" AND ").to_string()
                 } else {
                     "".to_string()
                 };
-                let statement = format!("SELECT channel_id, created, events FROM event_aggregates {} ORDER BY created DESC LIMIT {}", where_clause, limit);
+                let statement = format!(
+                    "
+                        WITH aggregates AS (
+                            SELECT 
+                                channel_id,
+                                created,
+                                event_type,
+                                jsonb_build_object(
+                                    'eventCounts',
+                                    jsonb_object_agg(
```

Comment:
how does this `agg` actually sum the values in counts/payouts?

Reply:
It doesn't `SUM` the values, what it does it return aggr JSON similar to the implementation in JS. We don't `SUM` the aggr values in JS either.  At first I tried `SUM` but realized we can't `SUM` because we use the `aggr.created` time and `SUM` query won't return that because it would be across multiple times.